pkg/packagekit/updates.jsx

```diff
@@ -158,6 +164,14 @@ function count_security_updates(updates) {
     return num_security;
 }
 
+function count_kpatch_updates(updates) {
+    let num_kpatches = 0;
+    for (const u in updates)
+        if (updates[u].name.startsWith("kpatch-patch"))
+            ++num_kpatches;
+    return num_kpatches;
+}
```

Comment:
I don't particularly like duplicating the "detect kpatch package name" logic here. updateItem() already has this:

    if (pkgNames.some(p => p.name.startsWith("kpatch-patch")))

and below you have another instance:

    ids = ids.filter(id => this.state.updates[id].name.startsWith("kpatch-patch"));

Can you please move this out into a `filterKpatchPackages()` helper function, that does the latter, and use it in all three places? I'm not sure how robust/reliable that prefix is, or whether e.g. other OSes (Ubuntu also supports kpatches, but in a slightly different form) also name them that way.

Reply:
I tried to find a better way to identify kpatches packages but as far as I can tell there is no constant defined in either PackageKit or DNF to identify kpatches only security/normal updates.

Reply:
Right, that's fine. I'd just like to have a single central function for it, instead of three places.

Reply:
The data is not entirely the same however, pkgNames is a list of dict's [{pkgname: "lala"}], this.state.updates is a dict.

Reply:
Not entirely sure I like my cooked up solution as it involves creating a dict from a list of dictionaries, so I don't have to deal with different data types in `filterKpatchPackages`.

Reply:
Right, that looks a bit ugly. Perhaps a simpler `isKpatchPackage(name) → bool` would be better?